src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -4373,43 +4373,38 @@ module Queries = struct
           Mina_base.Epoch_seed.to_base58_check
             next_epoch.Mina_base.Epoch_data.Poly.seed
         in
-        let%bind staking_ledger =
-          match Mina_lib.staking_ledger mina with
-          | None ->
-              Deferred.Result.fail "Staking ledger is not initialized."
-          | Some (Genesis_epoch_ledger l) ->
-              return (Ledger.Any_ledger.cast (module Ledger) l)
-          | Some (Ledger_db l) ->
-              return (Ledger.Any_ledger.cast (module Ledger.Db) l)
+        let mina_config = Mina_lib.config mina in
+        let local_state = mina_config.consensus_local_state in
+        let consensus_constants =
+          mina_config.precomputed_values.consensus_constants
+        in
+        let cast_ledger :
+               Consensus.Data.Local_state.Snapshot.Ledger_snapshot.t
+            -> Ledger.Any_ledger.witness = function
+          | Genesis_epoch_ledger l ->
+              Ledger.Any_ledger.cast (module Ledger) l
+          | Ledger_db l ->
+              Ledger.Any_ledger.cast (module Ledger.Db) l
+        in
+        let staking_ledger =
+          Consensus.Hooks.get_epoch_ledger ~constants:consensus_constants
```

Comment:
The PR description says that we are fetching the ledger corresponding to a particular block, but here there's no reference to the block at all. All parameters here are set at startup, and necessarily cannot infer which block.

Reply:
I disagree.

`~consensus_state:consensus` is dependent on a block and is used to extract epoch in function `get_epoch_ledger`